### DISCUSSION TITLE
Enhancing Keyboard Navigation for GitHub Actions

### DISCUSSION THREAD

**Discussion by megait004** [community/community#135150] [Accessibility]
### Select Topic Area

Question

### GitHub Feature Area

Actions

### Body

Hello everyone,

I'm reaching out to discuss and gather feedback on improving keyboard navigation for GitHub Actions. As a frequent user, I've noticed some challenges with accessibility, particularly when trying to navigate through workflows and logs using keyboard shortcuts alone.

Could we consider implementing more comprehensive keyboard shortcuts or enhancing the existing ones? This could greatly benefit users who rely on keyboard navigation due to mobility impairments or those who prefer using keyboards over mice.

Looking forward to your thoughts and any suggestions you might have!

Thank you.

**Comment by phungdat28** [comment]


**Reply by megait004** [reply]
Thank you for the thoughtful feedback and excellent suggestions regarding keyboard navigation improvements for GitHub Actions. Your ideas not only address the needs of users with mobility impairments but also enhance the efficiency for all power users who prefer keyboard controls. Here's a response to your suggestions:

Expanded Shortcut Coverage: Your idea to introduce shortcuts for navigating between workflow runs and jumping to specific log sections like errors and warnings is very insightful. This would indeed make the tool more efficient and user-friendly. Integrating shortcuts for toggling the display of elements like artifacts or environment variables could also streamline the user experience significantly.

Customizable Shortcuts: Allowing users to customize their keyboard shortcuts is a fantastic suggestion. It acknowledges the diverse needs and preferences of users, providing flexibility and improving accessibility for individuals with specific requirements.

Improved Focus Management: Ensuring clear focus states and streamlined tab orders is crucial for accessibility. Your emphasis on this shows a deep understanding of user interface design principles that cater to all users, especially those navigating primarily via keyboard.

ARIA Landmarks and Roles: Implementing ARIA landmarks and roles is indeed a critical step towards making GitHub Actions more accessible. This approach would greatly assist users who depend on assistive technologies like screen readers, enhancing their ability to navigate and use the platform effectively.

Quick Access Panels: The suggestion to introduce a keyboard-activated command palette is an innovative solution that mirrors successful implementations in other development tools like Visual Studio Code. This feature would speed up access to commonly used functions and could significantly boost productivity.

Feedback Loops: Incorporating a direct feedback mechanism within GitHub Actions for accessibility-related issues is a proactive strategy. It would help in continuously improving the platform based on user input, ensuring that the tool evolves to meet the needs of its diverse user base.

User Testing: Engaging with users who rely on keyboard navigation for user testing is an excellent point. Real-world testing can provide invaluable insights that can be used to refine and optimize keyboard navigation further.

Your contributions are incredibly valuable to the community, and I hope to see some of these enhancements implemented, making GitHub Actions more accessible and efficient for everyone. Thank you for sharing your ideas, and let's continue to push for a more inclusive and user-friendly platform.

**Reply by vugiap04** [reply]
Absolutely, your commitment to enhancing accessibility in GitHub Actions is greatly appreciated! Your detailed suggestions, such as expanding shortcut coverage and introducing customizable shortcuts, are particularly compelling and could significantly enhance the efficiency and accessibility of the platform for all users. The proposal to implement ARIA roles for better screen reader compatibility and the innovative idea of a quick access command palette like in Visual Studio Code could transform user interactions with GitHub Actions. Engaging in user testing with a focus on keyboard navigation will undoubtedly lead to meaningful improvements based on real user feedback. It's exciting to see these ideas being proposed and discussed, and I hope they gain the necessary attention and implementation. Thank you for your valuable contributions to making GitHub a more inclusive platform!

**Comment by gzh142410** [comment]
@megait004 very good

**Reply by megait004** [reply]
Absolutely, enhancing keyboard navigation in GitHub Actions is an excellent initiative, benefiting not only users with mobility impairments but also power users who favor keyboard shortcuts. Your suggestions are insightful and could significantly improve user experience. Expanding shortcut coverage, allowing for customizable shortcuts, and implementing ARIA roles to assist screen reader users are particularly pivotal. The idea of a command palette for quick access to common actions is also a standout suggestion. Engaging with the user community through testing and feedback will ensure that these improvements meet real-world needs. Thank you for pushing for greater accessibility and efficiency in GitHub Actions!

**Reply by gzh142410** [reply]
great

**Reply by giaphue0404** [reply]
@megait004 
Great suggestions! Enhancing keyboard navigation in GitHub Actions not only boosts accessibility but also improves overall user efficiency. Your ideas for customizable shortcuts and a command palette are spot-on and could really streamline workflows. Thanks for advocating for these important improvements!

**Comment by Hue041004** [comment]
@megait004 
Fantastic insights! Your focus on enhancing keyboard navigation for GitHub Actions is truly commendable. Introducing more intuitive shortcuts and better focus management would undoubtedly make the platform more accessible and user-friendly. Thanks for sharing these great ideas!

**Comment by queenofcorgis** [comment]
@megait004 @vugiap04 @phungdat28 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
